### PR TITLE
fix(training): kcrps losses handling NaNs for validation

### PR DESCRIPTION
**Pull request by sahahner**
## Description

All our losses have the option `ignore_nans` as we want to ignore NaN fields if the losses are used as validation metrics. 
This option is still missing for the kcrps losses. 

Instead of using `torch.sum` and `torch.mean` use the sum and avg functions of the loss class to allow or not allow NaNs depending on what has been selected in the config file.

## Type of Change

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation update

## Code Compatibility

-   [x] I have performed a self-review of my code

### Code Performance and Testing

-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I ran the [complete Pytest test](https://anemoi.readthedocs.io/projects/training/en/latest/dev/testing.html) suite locally, and they pass
-   [ ] I have tested the changes on a single GPU
-   [ ] I have tested the changes on multiple GPUs / multi-node setups
-   [ ] I have run the Benchmark Profiler against the old version of the code
-   [ ] If the new feature introduces modifications at the config level, I have made sure to update Pydantic Schemas and default configs accordingly

### Dependencies

-   [ ] I have ensured that the code is still pip-installable after the changes and runs
-   [ ] I have tested that new dependencies themselves are pip-installable.
-   [ ] I have not introduced new dependencies in the inference portion of the pipeline

### Documentation

-   [ ] My code follows the style guidelines of this project
-   [ ] I have updated the documentation and docstrings to reflect the changes
-   [ ] I have added comments to my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by ssmmnn11** [comment]
Thank you for the pull request. 

There is always a question how to handel NaNs. I think the best way is to do it like in the other losses, in the spatial aggregation part. Here, the NaN "handling" happens in the point wise crps score computation, which leads to undefined behaviour if different ensemble members had NaNs at different places - though this is currently not the case. 

The loss scaling is currently broken after https://github.com/ecmwf/anemoi-core/pull/52. My suggestion is that we wait until this is fixed (hopefully soon) and then we can implement NaN handling in a consistent way in all losses.
